src/sql/workbench/services/connection/browser/connectionBrowseTab.ts

```diff
@@ -79,24 +137,52 @@ export class ConnectionBrowserView extends Disposable implements IPanelView {
 		];
 
 		this.model = this.instantiationService.createInstance(TreeModel);
-
+		this.treeDataSource = new DataSource();
 		this.tree = this._register(this.instantiationService.createInstance(
 			WorkbenchAsyncDataTree,
 			'Browser Connections',
-			container,
+			this.treeContainer,
 			new ListDelegate(),
 			renderers,
-			new DataSource(),
+			this.treeDataSource,
 			{
 				identityProvider: new IdentityProvider(),
 				horizontalScrolling: false,
 				setRowLineHeight: false,
 				transformOptimization: false,
 				accessibilityProvider: new ListAccessibilityProvider()
 			}) as WorkbenchAsyncDataTree<TreeModel, TreeElement>);
-
+		this.tree.onContextMenu(e => {
+			const context = e.element as ITreeItemFromProvider;
+			if (context?.element) {
+				this.contextKey.set(context.element);
+				const menu = this.menuService.createMenu(MenuId.ConnectionDialogBrowseTreeContext, this.contextKeyService);
+				const primary: IAction[] = [];
+				const secondary: IAction[] = [];
+				const result = { primary, secondary };
+				createAndFillInContextMenuActions(menu, { shouldForwardArgs: true }, result, this.contextMenuService);
+
+				this.contextMenuService.showContextMenu({
+					getAnchor: () => e.anchor,
+					getActions: () => result.primary,
+					getActionsContext: () => (<TreeViewItemHandleArg>{ $treeViewId: context.treeId, $treeItemHandle: context.element.handle, $treeItem: context.element })
+				});
+			}
+		});
 		this.tree.onMouseDblClick(e => this._onDblClick.fire(e));
 		this.tree.onMouseClick(e => this._onSelect.fire(e));
+		this.tree.onDidOpen((e) => {
+			if (!e.browserEvent) {
+				return;
+			}
+			const selection = this.tree.getSelection();
+			if (selection.length === 1) {
+				const selectedNode = selection[0];
+				if ('element' in selectedNode && selectedNode.element.command) {
```

Comment:
Would be good to at least provide a link where this is set and/or a brief description of what this command is expected to be

Reply:
the selectedNode.element is not an type we defined here, it comes from vscode:https://github.com/microsoft/azuredatastudio/blob/6ff1e3866bcad2a8a639fdaf6e6fe4aeab49b2b2/src/vs/vscode.d.ts#L8499